CMSgov/cmcs-eregulations
# EREGCSC-1338 pagination controls

Resolves [EREGCSC-1338](https://jiraent.cms.gov/browse/EREGCSC-1338)

**Description**

Add pagination controls to bottom of resources search page when there are results being displayed.

**This pull request changes**

- Formats a few existing files using prettier but makes no other changes to them (ex: `SearchEmptyState`)
- Promotes floating nav components from prototype
   - disables storybook stories for these new components because they are not yet working with our prod bundling process
   - renames `FloatingNavBtn` to be `NavBtn` to represent more generalized usage, not just in bottom/floating nav
- Creates `PaginationController` that is generic enough to be used in any single page app view that uses the Vue Router
- Styles pagination controls to resemble [CMS Design System Pagination Component](https://design.cms.gov/storybook/?path=/story/components-pagination--default&globals=theme:core).  See JIRA ticket above to see list of design differences
  - On desktop it should resemble "Default Navigation" in the CMS Design System link above
  - On narrow screens (`mobile-max` screen widths and below, ~768px) it should resemble "Compact Navigation" in the CMS Design System link above
  - adds `page` URL query parameter to keep track of page state
  - Loads all 2000+ resources on Resources view load and allows users to sort them and paginate through them without adding any additional filters

**Steps to manually verify this change:**

1. Visit resources page
2. Scroll to bottom of results
3. Click Previous, Next, and Page Number links
4. Take note of disabled states for Previous and Next buttons based on currently selected page
5. Copy link to deep page into new browser tab to ensure it loads expected page
6. Ensure resources page loads 2000+ resources on initial page load and that you can paginate/sort them
7. Ensure that when you are on a deep page (not page 1) and you change filters or search query that the results return you to page 1
